corehq/apps/data_interfaces/views.py

```diff
@@ -1150,21 +1171,38 @@ def post(self, request, *args, **kwargs):
         rule_params, action_params = self.parse_params(request)
         errors = self.validate_action_params(action_params)
         errors.extend(self.validate_rule_params(request.domain, rule_params, self.rule))
+
+        cases_filter_form = DedupeCaseFilterForm(request.domain, request.POST)
+
+        if not cases_filter_form.is_valid():
+            errors.extend(cases_filter_form.errors)
+
         if errors:
-            error_message = _("Deduplication rule not saved. ")
-            messages.error(request, error_message + "; ".join(errors))
+            messages.error(
+                request, _('Deduplication rule not saved due to the following issues: %s') %
+                '<ul><li>{}</li></ul>'.format('</li><li>'.join(errors)),
+                extra_tags='html'
+            )
+
             kwargs.update({
                 'error': True,
                 'rule_params': rule_params,
                 'action_params': action_params,
             })
             return self.get(request, *args, **kwargs)
 
+        filter_criteria_updated = False
+
         with transaction.atomic():
             rule_modified = self._update_model_instance(self.rule, rule_params)
             action_modified = self._update_model_instance(self.dedupe_action, action_params)
 
-        if rule_modified or action_modified:
+            # Is there a good way to check if the criteria has changed without
+            # going over each and every value?
+            cases_filter_form.save_criteria(self.rule, save_meta=False)
+            filter_criteria_updated = True
```

Comment:
`form.save_criteria()` overwrites the criteria definitions by wiping the current ones and reconstructing new ones based on the form cleaned_data...but this leaves us without knowing if there ever was a change made...I was thinking of rewriting that method (or writing a new one) so that we can know if any criteria was actually updated, but going through the current criteria and comparing it to the new ones seems...sub-optimal...but I could be mistaken.

Any thoughts?